### DISCUSSION TITLE
ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by moon47usaco** [TheLastBen/fast-stable-diffusion#1708] [General]
This Colab has been my go to for the last week. best around. =]

Getting ModuleNotFoundError.

Did not have this problem for the past week, was fine last night. 

Tried to remove the folder on GDrive and start over, same.

If I try to pip install fastapi in the colab terminal it complains no gradio, etc...

`Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 7, in <module>
    from fastapi import FastAPI
ModuleNotFoundError: No module named 'fastapi'`

**Comment by CreativeTime** [comment]
i have the same problem

**Comment by TheLastBen** [comment]
colab changed to python 3.9, I'll update the dependencies, it'll take some time

**Reply by CreativeTime** [reply]
very thank you. i´ll wait
have a nice day

**Reply by CreativeTime** [reply]
Where will you let us know when it's ready?

**Reply by TheLastBen** [reply]
by checking the latest commit

**Reply by moon47usaco** [reply]
Constantly a moving target... =]

Thank you.

**Reply by royalsg22** [reply]
Thank You @TheLastBen

**Reply by meunumerotim333** [reply]
@TheLastBen Can you update for me on this old Colab too?

https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb

This one is working perfect for me on my values. I wish to continue to use this one.

Or when you update all the old collabs will auto-update too?

**Comment by SDPDVADER** [comment]
Thank you @thelastben!

**Comment by Headupdef** [comment]
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 7, in <module>
    from fastapi import FastAPI
ModuleNotFoundError: No module named 'fastapi'

**Reply by Headupdef** [reply]
thanks for working on it!

**Comment by bayoumedic** [comment]
i cant take it.  i need a fix.  ill post a prompt if someone googles it and posts the pic.   cool?

**Reply by CreativeTime** [reply]
!pip install "fastapi[all]"
!pip install "gradio[all]"
!pip install "lpips[all]"
!pip install "basicsr[all]"
!pip install GitPython
!pip install safetensors
!pip install omegaconf
!pip install transformers
!pip install open_clip
!pip install diffusers
!pip install triton
!pip install bitsandbytes
!pip install wget
!pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements_versions.txt
!pip install open_clip_torch
!pip install git+https://github.com/openai/CLIP.git
!pip install xformers

**Reply by SaturnoElMago** [reply]
This is the solution for this issue @CreativeTime or what is it?

**Reply by moon47usaco** [reply]
If you have a paid colab subscription you can paste that into the command line (lower left most icon) for the runtime.

@CreativeTime Thank you for a quick fix while we wait for @TheLastBen to update the page.

Confirmed working for me.

I need to paste them in without "!" and do not forget to hit enter for the last one so it executes... =]

pip install "gradio[all]"
pip install "lpips[all]"
pip install "basicsr[all]"
pip install GitPython
pip install safetensors
pip install omegaconf
pip install transformers
pip install open_clip
pip install diffusers
pip install triton
pip install bitsandbytes
pip install wget
pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements_versions.txt
pip install open_clip_torch
pip install git+https://github.com/openai/CLIP.git
pip install xformers

**Reply by SaturnoElMago** [reply]
@moon47usaco  
didnt work for me, I dont know why.
This is the error appearing after running it.

Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 15, in <module>
    from modules import import_hook, errors, extra_networks, ui_extra_networks_checkpoints
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/ui_extra_networks_checkpoints.py", line 6, in <module>
    from modules import shared, ui_extra_networks, sd_models
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 17, in <module>
    from modules.sd_hijack_inpainting import do_inpainting_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack_inpainting.py", line 4, in <module>
    from einops import repeat
ModuleNotFoundError: No module named 'einops'

**Reply by moon47usaco** [reply]
Try: running all forms except the run SD (last one)
Then paste that in terminal, be sure xformers installs. 
IE wait for everything to load in terminal and press enter one last time for xformers.
Run the final form.

Other than that I am not sure.

**Comment by SaturnoElMago** [comment]
OMG! Actually working 100% Thank you @moon47usaco and @CreativeTime, great solution while we wait for @TheLastBen to update it.

**Reply by meunumerotim333** [reply]
Can you share your Fix colab that is working?

**Reply by SaturnoElMago** [reply]
Its not working by itself, I constantly need to make the terminal execution that @moon47usaco and @CreativeTime made out.

pip install "gradio[all]"
pip install "lpips[all]"
pip install "basicsr[all]"
pip install GitPython
pip install safetensors
pip install omegaconf
pip install transformers
pip install open_clip
pip install diffusers
pip install triton
pip install bitsandbytes
pip install wget
pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements_versions.txt
pip install open_clip_torch
pip install git+https://github.com/openai/CLIP.git
pip install xformers

**Reply by Morgan2001** [reply]
and if there is no paid subscription, then do not use this fix in any way? just wait until they fix it?

**Reply by astragor** [reply]
> and if there is no paid subscription, then do not use this fix in any way? just wait until they fix it?

Yes you can fix it even if you use a free plan.
You have to add a new code and copy and execute the code given by @CreativeTime.
And rerun the SD part of th script after

**Comment by TheLastBen** [comment]
fixed, update your colab

**Reply by AiXplorer777** [reply]
Thanks @TheLastBen ! You're the best! :)

**Comment by CreativeTime** [comment]
somebody knows if alredy it works.?

**Reply by meunumerotim333** [reply]
Yes, its working, but's buggy.

**Reply by CreativeTime** [reply]
with the fix or the error was solved by last ben?
